src/upstream/the_graph/mod.rs

```diff
@@ -81,6 +83,33 @@ struct DomainEvent {
     transaction_id: String,
 }
 
+fn choose_endpoint() -> String {
+    let mut options: Vec<String> = Vec::new();
+
+    if let Some(value) = C.upstream.the_graph.subgraph0.clone() {
+        options.push(value);
+    }
+    if let Some(value) = C.upstream.the_graph.subgraph1.clone() {
+        options.push(value);
+    }
+    if let Some(value) = C.upstream.the_graph.subgraph2.clone() {
+        options.push(value);
+    }
+    if let Some(value) = C.upstream.the_graph.subgraph3.clone() {
+        options.push(value);
+    }
+    if let Some(value) = C.upstream.the_graph.subgraph4.clone() {
+        options.push(value);
+    }
```

Comment:
OK for now. May refactor into `Vec<String>` later (using func like  `.sample()` in Ruby)